cmd/kube-proxy/app/server.go

```diff
@@ -186,14 +188,19 @@ func (o *Options) Complete() error {
 		o.applyDeprecatedHealthzPortToConfig()
 	}
 
+	err := utilfeature.DefaultFeatureGate.SetFromMap(o.config.FeatureGates)
```

Comment:
As long as this is after the flags have been parsed into the config object, it's probably ok.

You'll probably also want to set the gates after loading the config file, if you allow feature gates to be set in the file.

Read over the flag precedence issue we had with the Kubelet and see if it also applies to you: #56171. If you need flags to take precedence, you'll likely also want to ensure that feature gates accumulate between the file and the CLI, by setting gates first from the file and then again from the CLI.

Reply:
This would be ok as per folks' discussion: https://github.com/kubernetes/kubernetes/pull/55234#issuecomment-364141697, right?

Reply:
I think so, yes.

Reply:
Given that we apply the feature gates from `o.config.FeatureGates` in both scenarios (flags and config file), how about we only call this once, at the end of this function (and remove the call in the if block on line 203)?

Reply:
Since you intend for none of the flags to be honored when a file is loaded, this should be ok.
Otherwise you'd want to ensure additive behavior, e.g. flags have gate a=true, and file has gate b=true, and you get a=true,b=true at the end of the day.

Reply:
Updated.